backend/models/sequenceActivities.js

```diff
@@ -0,0 +1,298 @@
+	activities = getSubActivities(null, foundActivities);
+
+	return { activities };
+};
+
+const SequenceActivities = {};
+
+SequenceActivities.getSequenceActivityDetail = async (account, model, sequenceId, activityId) => {
+	const sequence = await db.findOne(account, model + ".sequences", { _id: utils.stringToUUID(sequenceId)});
+
+	if (!sequence) {
+		throw responseCodes.SEQUENCE_NOT_FOUND;
+	}
+
+	const activity = await db.findOne(account, activityCol(model), {"_id": utils.stringToUUID(activityId)});
+
+	if (!activity) {
+		throw responseCodes.ACTIVITY_NOT_FOUND;
+	}
+
+	return cleanActivityDetail(activity);
+};
+
+SequenceActivities.create = async (account, model, sequenceId, activity) => {
+	const sequence = await db.findOne(account, model + ".sequences", { _id: utils.stringToUUID(sequenceId)});
+
+	if (!sequence) {
+		throw responseCodes.SEQUENCE_NOT_FOUND;
+	}
+
+	if (!activitySchema.isValidSync(activity, { strict: true })) {
+		throw responseCodes.INVALID_ARGUMENTS;
+	}
+
+	if (activity.parent && !await db.findOne(account,  activityCol(model), { _id: utils.stringToUUID(activity.parent)})) {
+		throw responseCodes.INVALID_ARGUMENTS;
+	}
+
+	const _id = nodeuuid();
+	activity = {...activity, sequenceId:  utils.stringToUUID(sequenceId), _id: utils.stringToUUID(_id)};
+
+	await Promise.all([
+		db.insert(account,  activityCol(model), activity),
+		FileRef.removeFile(account, model, "activities", sequenceId)
+	]);
+
+	return {...activity, _id, sequenceId: sequenceId};
+};
+
+SequenceActivities.edit = async (account, model, sequenceId, activityId, activity) => {
+	const sequence = await db.findOne(account, model + ".sequences", { _id: utils.stringToUUID(sequenceId)});
+
+	if (!sequence) {
+		throw responseCodes.SEQUENCE_NOT_FOUND;
+	}
+
+	if (!activityEditSchema.isValidSync(activity, { strict: true })) {
+		throw responseCodes.INVALID_ARGUMENTS;
+	}
+
+	if (activity.parent && !await db.findOne(account, activityCol(model), { _id: utils.stringToUUID(activity.parent)})) {
+		throw responseCodes.INVALID_ARGUMENTS;
+	}
+
+	const query =  {_id: utils.stringToUUID(activityId), sequenceId: utils.stringToUUID(sequenceId)};
+	const {result} = await db.update(account, activityCol(model), query, {$set: activity});
+
+	if (!result.n) {
+		throw responseCodes.ACTIVITY_NOT_FOUND;
+	}
+
+	await FileRef.removeFile(account, model, "activities", sequenceId);
+};
+
+SequenceActivities.remove = async (account, model, sequenceId, activityId) => {
+	const sequence = await db.findOne(account, model + ".sequences", { _id: utils.stringToUUID(sequenceId)});
+
+	if (!sequence) {
+		throw responseCodes.SEQUENCE_NOT_FOUND;
+	}
+	const activities = await db.find(account,activityCol(model), {sequenceId: utils.stringToUUID(sequenceId), parent:{$exists: true}} , {parent:1, _id:1});
+
+	const idsToDelete = getDescendantsIds(activities, activityId);
```

Comment:
I guess this may change anyway since we're going to store children instead of parents.

But I think it's better to just let the db to do the query (even if it means we have to do it multiple times with different parents)

i.e. just do `parent: activityId`